server/core/src/irodsReServer.cpp

```diff
@@ -77,6 +78,52 @@ namespace {
         }
     }
 
+    /// Load the available executors from the server's configuration
+    void load_available_executors(){
+        logger::delay_server::info("Picking out available executors");
+        if(irods::advanced_setting_exists(irods::HOSTS_AVAILABLE_RULE_EXECUTORS)){
+            auto &hosts = irods::get_advanced_setting<std::vector<boost::any> >(irods::HOSTS_AVAILABLE_RULE_EXECUTORS);
+            for(const auto &i : hosts){
+                available_hosts.push_back(boost::any_cast<std::string>(i));
+            }
+        }
+        rodsEnv env;
+        _getRodsEnv(env);
+        if(available_hosts.empty()){
+            available_hosts.push_back(env.rodsHost);
+        }
+        logger::delay_server::info("Executors populated");
+    }
+
+    /// Get the next connection to use for the next rule
+    ///
+    /// \param[in] user The name of the user to proxy into
+    ix::client_connection get_next_connection(const std::optional<std::string>& _user){
```

Comment:
this function name may need to be tweaked. it is returning a new connection every time it is invoked.

imo, having **next** in the name makes it a bit confusing. perhaps `get_connection` or `new_connection`?

Reply:
either seems fine.   or combined... `get_new_connection`?